plenum/test/restart/test_persist_pp.py

```diff
@@ -0,0 +1,134 @@
+           node_status_db_serializer.deserialize(
+               primary_replicas_new[2].node.nodeStatusDB.get(LAST_SENT_PRE_PREPARE))[2]
+
+
+def test_clear_persisted_last_pp_after_view_change(looper, txnPoolNodeSet, tconf, tdir,
+                                                   sdk_pool_handle, sdk_wallet_client, allPluginsPath):
+    sdk_send_batches_of_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                                         sdk_wallet_client, batches_count, batches_count)
+
+    primary_replicas = get_primary_replicas(txnPoolNodeSet)
+    old_3pc = primary_replicas[0].last_ordered_3pc
+    # Restart master primary to make a view_change
+    restart_nodes(looper, txnPoolNodeSet, [primary_replicas[0].node], tconf, tdir, allPluginsPath,
+                  after_restart_timeout=tconf.ToleratePrimaryDisconnection + 1)
+    primary_replicas_new = get_primary_replicas(txnPoolNodeSet)
+    assert primary_replicas[0] not in primary_replicas_new
+
+    # Check that we've cleared last send pre-prepare on every primary replica
+    assert LAST_SENT_PRE_PREPARE not in primary_replicas_new[1].node.nodeStatusDB
+
+    sdk_send_batches_of_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                                         sdk_wallet_client, batches_count, batches_count)
+    persisted = node_status_db_serializer.deserialize(
+        primary_replicas_new[1].node.nodeStatusDB.get(LAST_SENT_PRE_PREPARE))
+    assert persisted[2] == batches_count
+    assert persisted[1] == old_3pc[0] + 1
+
+
+def test_clear_persisted_last_pp_after_pool_restart(looper, txnPoolNodeSet, tconf, tdir,
+                                                    sdk_pool_handle, sdk_wallet_client):
+    sdk_send_batches_of_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle,
+                                         sdk_wallet_client, batches_count, batches_count)
+
+    for node in txnPoolNodeSet:
+        node.stop()
+        looper.removeProdable(node)
+
+    # Starting nodes again by creating `Node` objects since that simulates
+    # what happens when starting the node with script
+    restartedNodes = []
+    for node in txnPoolNodeSet:
+        config_helper = PNodeConfigHelper(node.name, tconf, chroot=tdir)
+        restartedNode = TestNode(node.name,
+                                 config_helper=config_helper,
+                                 config=tconf, ha=node.nodestack.ha,
+                                 cliha=node.clientstack.ha)
+        looper.add(restartedNode)
+        restartedNodes.append(restartedNode)
+
+    looper.run(checkNodesConnected(restartedNodes))
+    ensureElectionsDone(looper, restartedNodes)
+    checkViewNoForNodes(restartedNodes, 0)
+
+    # Check that we've cleared last send pre-prepare on every primary replica
+    primary_replicas = get_primary_replicas(restartedNodes)
+    assert primary_replicas[0].last_ordered_3pc == (0, 0) and \
+           primary_replicas[0].lastPrePrepare is None
```

Comment:
We should check `lastPrePrepareSeqNo` here rather than `lastPrePrepareSeq`.

Reply:
fixed